docs/source/existing_code.rst

```diff
@@ -0,0 +1,174 @@
+.. _existing-code:
+
+Using mypy with an existing codebase
+====================================
+
+This section explains how to get started using with an existing,
+significant codebase that has no or few type annotations. If you are
+a beginner, you can skip this section.
+
+These steps will get you started with mypy on an existing codebase:
+
+1. Start small -- get a clean mypy build for some files, with few
+   annotations
+
+2. Write a mypy runner script to ensure consistent results
+
+3. Run mypy in Continuous Integration to prevent type errors
+
+4. Gradually annotate commonly imported library modules
+
+5. Write annotations as you modify existing code and write new code
+
+6. Use MonkeyType or PyAnnotate to automatically annotate legacy code
+
+We discuss all of these points in some detail below, and a few optional
+follow-up steps.
+
+Start small
+-----------
+
+If your codebase is large, pick a subset of your codebase (say, 5,000
+to 50,000 lines) and run mypy only on this subset at first,
+*without any annotations*. This shouldn't take more than a day or two
+to implement, so you start enjoying benefits soon.
+
+You'll likely need to fix some mypy errors, such as by inserting
+annotations requested by mypy, or by adding ``# type: ignore``
+comments to silence errors you don't want to fix now.
+
+In particular, mypy often generates errors about modules that it can't
+find or that don't have stub files:
+
+.. code-block:: text
+
+    core/config.py:7: error: Cannot find module named 'frobnicate'
+    core/model.py:9: error: Cannot find module named 'acme'
+    ...
+
+This is normal, and you can easily ignore these errors. For example,
+here we ignore an error about a third-party module ``frobnicate`` that
+doesn't have stubs using ``# type: ignore``:
+
+.. code-block:: python
+
+   import frobnicate  # type: ignore
+   ...
+   frobnicate.initialize()  # OK (but not checked)
+
+You can also use a mypy configuration file, which is convenient if
+there are a large number of errors to ignore. For example, to disable
+errors about importing ``frobnicate`` and ``acme`` everwhere in your
+codebase, use a config like this:
+
+.. code-block:: text
+
+   [mypy-frobnicate.*]
+   ignore_missing_imports = True
+
+   [mypy-acme.*]
+   ignore_missing_imports = True
+
+You can add multiple sections for different modules that should be
+ignored.
+
+If your config file is named ``mypy.ini``, this is how you run mypy:
+
+.. code-block:: text
+
+   mypy --config-file mypy.ini mycode/
+
+If you get a large number of errors, you may want to ignore all errors
+about missing imports.  This can easily cause problems later on and
+hide real errors, and it's only recommended as a last resort. We
+don't explain this option in more detail here.
```

Comment:
The last sentence: maybe instead just say "More details here." and just link to the relevant section in the command line docs?

(The only downside is that the command line docs don't really do a good job of explaining the cons of the `--ignore-missing-imports` flag, but that's probably something we can fix in a future PR.)